src/geoapis/vector.py

```diff
@@ -12,27 +12,34 @@
 import geopandas
 
 
-class Linz:
-    """ A class to manage fetching Vector data from LINZ.
+class WfsQueryBase:
+    """ An abstract class to manage fetching Vector data using WFS.
 
-    API details at: https://www.linz.govt.nz/data/linz-data-service/guides-and-documentation/wfs-spatial-filtering
+    API details at: https://www.ogc.org/standards/wfs or
+    https://www.linz.govt.nz/data/linz-data-service/guides-and-documentation/wfs-spatial-filtering
 
-    The specified vector layer is queried each time run is called and any vectors passing though the catchment defined
-    in the bounding_polygon are returned.
+    The specified vector layer is queried each time run is called, and any layer features passing though the optionally
+    defined bounding_polygon are returned. If no bounding_polygon is specified all layer features are returned.
 
     Flexibility exists in the inputs. Only the key is required. If no bounding_polygon is specified all features in a
     layer will be downloaded. If no crs is specified, the bounding_polygon will be used if the bounding_polygon is
     specified. If no CRS or bounding_polygon is specified the CRS of the downloaded features will be used. """
 
+    @property
+    def NETLOC_API():
+        raise NotImplementedError
+
+    @property
+    def GEOMETRY_NAME():
+        raise NotImplementedError
```

Comment:
You could make these abstract methods using ABC (abstract base class) module's @abstract method and making the class an ABC too

Reply:
Cool - I will do that.

Reply:
Great - I've done that now! looks better for sure. Harder to forget to implement as well.